app/assets/javascripts/preferences/panes/account/deleteAccountDialog/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { Button } from '@/components/Button';
+import { useState } from 'preact/hooks';
+import { DeleteAccountForm } from '@/preferences/panes/account/deleteAccountDialog/DeleteAccountForm';
+import {
+  ModalDialog,
+  ModalDialogButtons,
+  ModalDialogDescription,
+  ModalDialogLabel
+} from '@/components/shared/ModalDialog';
+import { DeleteAccountSubmitted } from '@/preferences/panes/account/deleteAccountDialog/DeleteAccountSubmitted';
+import { executeCallbackWhenEnterIsPressed } from '@/utils';
+import { FunctionalComponent } from 'preact';
+import { JSXInternal } from '@node_modules/preact/src/jsx';
+import TargetedKeyboardEvent = JSXInternal.TargetedKeyboardEvent;
+import { ErrorMessages } from '@/enums';
+import { WebApplication } from '@/ui_models/application';
+import { User } from '@node_modules/@standardnotes/snjs/dist/@types/services/api/responses';
+
+enum SubmitButtonTitles {
+  Default = 'Delete my account for good',
+  Finish = 'Finish'
+}
+
+enum Steps {
+  InitialStep,
+  FinishStep
+}
+
+type Props = {
+  onCloseDialog: () => void;
+  application: WebApplication;
+}
+
+export const DeleteAccountDialog: FunctionalComponent<Props> = ({
+  onCloseDialog,
+  application,
+}) => {
+  const [submitButtonTitle, setSubmitButtonTitle] = useState(SubmitButtonTitles.Default);
+  const [currentStep, setCurrentStep] = useState(Steps.InitialStep);
+  const [password, setPassword] = useState('');
+  const [isRequestInProgress, setIsRequestInProgress] = useState(false);
+
+  const { EnterPassword, IncorrectPassword, SomethingWentWrong } = ErrorMessages;
+  const { alert: snAlert } = application.alertService;
+
+  const handleKeyPress = (event: TargetedKeyboardEvent<HTMLInputElement>) => {
+    executeCallbackWhenEnterIsPressed(event.key, handleSubmit);
+  };
+
+  const cleanupBeforeCancelSubmit = () => {
+    setIsRequestInProgress(false);
+  };
+
+  const handleSubmit = async () => {
+    if (isRequestInProgress) {
+      return;
+    }
+
+    if (currentStep === Steps.FinishStep) {
+      onCloseDialog();
+      return;
+    }
+
+    setIsRequestInProgress(true);
+
+    if (password === '') {
```

Comment:
```suggestion
    if (password.trim() === '') {
```

Reply:
I implemented it this way, but then removed, since I thought specifically for passwords we don't need trimming, as the user might put space in their password. I know it's not happening every day, but it's something we might consider, wdyt?

Reply:
Ah true 👍